File: ECMWF_downloader/ECMWF_ERA-Int_pl_downloader.py
#!/usr/bin/env python

import os
import datetime
from ecmwfapi import ECMWFDataServer
# if libs not installed, then:
# ! pip install https://software.ecmwf.int/wiki/download/attachments/56664858/ecmwf-api-client-python.tgz

#data_i = "20110501"
#data_f = "20110701"

dt = datetime.datetime(2006, 8, 18)
end = datetime.datetime(2006, 8, 21)
step = datetime.timedelta(days=1)

date_list = []

while dt <= end:
    date_list.append(dt.strftime('%Y%m%d'))
    dt += step


server = ECMWFDataServer()

for date in date_list :
    if os.path.isfile("ERA-Int_pl" + str(date) + ".grib") :
        print("ERA-Int_pl" + str(date) + ".grib file allready exists")

    else:
        server.retrieve({
            'levelist'  : "all",
            'levtype'   : "pl",
            'param'     : "129/130/131/132/133/157",
            'dataset'   : "interim",
            'step'      : "0",
            'grid'      : "128", # "0.75/0.75",   "0.125/0.125",   "128"
            'time'      : "00/06/12/18",
            'area'      : "42/333/22/353",
            'date'      : str(date) + "/to/" + str(date),
            'type'      : "an",
            'class'     : "ei",
            'stream'    : "oper",
            #'format'    : "netcdf",
            'target'    : "ERA-Int_pl" + str(date) + ".grib"

        })

File: ECMWF_downloader/ecmwf-api-client-perl/ECMWF_ERA-Int_pl_downloader.pl
#!/usr/bin/env perl

BEGIN { $ENV{PERL_LWP_SSL_VERIFY_HOSTNAME} = 0 }

use ECMWF::DataServer;

my $client = ECMWF::DataServer->new();

$client->retrieve(
	dataset => "interim",
	step    => "0",
#	number  => "all",
	levtype => "pl",
	levelist=> "all",
	date    => "20110501/to/20110701",
	time    => "00/06/12/18",
#	origin  => "all",
	type    => "an",
	param   => "129/130/131/132/157",
#	area    => "70/-130/30/-60",
	grid    => "128",
	target  => "ERA-Int_pl_.grib",
);
